scripts/spectrum-tokens.js

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import fg from 'fast-glob';
+import fs from 'fs-extra';
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+const tokensRoot = path.join(
+    __dirname,
+    '..',
+    'node_modules',
+    '@spectrum-css',
+    'tokens',
+    'dist',
+    'css',
+    '**',
+    '*.css'
+);
+
+const tokenPackages = ['actionbutton'];
+
+const packagePaths = tokenPackages.map((packageName) => {
+    return path.join(
+        __dirname,
+        '..',
+        'node_modules',
+        '@spectrum-css',
+        packageName,
+        'dist',
+        'themes'
+    );
+});
+
+const targetHost = (css) => {
+    css = css.replaceAll(/.spectrum {/g, ':host {');
+    css = css.replaceAll(/.spectrum--express {/g, ':host {');
+    css = css.replaceAll(
+        /^.spectrum--light, .spectrum--lightest {/g,
+        ':host {'
+    );
+    css = css.replaceAll(/.spectrum--dark {/g, ':host {');
+    css = css.replaceAll(/.spectrum--darkest {/g, ':host {');
+    css = css.replaceAll(/.spectrum--darkest {/g, ':host {');
+    css = css.replaceAll(/.spectrum--express.spectrum--medium {/g, ':host {');
+    css = css.replaceAll(/.spectrum--express.spectrum--large {/g, ':host {');
+    css = css.replaceAll(/.spectrum--medium {/g, ':host {');
+    css = css.replaceAll(/.spectrum--large {/g, ':host {');
+
+    return css;
+};
+
+const processTokens = (srcPath) => {
+    let css = fs.readFileSync(srcPath, 'utf8');
+    const fileName = srcPath.split(path.sep + 'css' + path.sep).at(-1);
+    css = targetHost(css);
+
+    fs.writeFileSync(
+        path.join(__dirname, '..', 'packages', 'styles', 'tokens', fileName),
+        css
+    );
+};
+
+const processPackages = async (srcPath, index) => {
+    const packageName = tokenPackages[index];
+    const expressPath = path.join(srcPath, 'express.css');
+    const spectrumPath = path.join(srcPath, 'spectrum.css');
+    let express = fs.readFileSync(expressPath, 'utf8');
+    let spectrum = fs.readFileSync(spectrumPath, 'utf8');
+    express = targetHost(express);
+    spectrum = targetHost(spectrum);
+
+    fs.appendFileSync(
+        path.join(
+            __dirname,
+            '..',
+            'packages',
+            'styles',
+            'tokens',
+            'express',
+            'global-vars.css'
+        ),
```

Comment:
Honestly, I thought it was you that told me _not_ to do this at one point 🤣. I'm OK with the long form and how it normalized to other usage in this file where it'll look like this and one of the directories will be a variable.

Reply:
snip, snap, snip, snap!

I think that was probably `${__dirname}/foo` (since I used to spend so much time debugging node apps written by windows users and deployed to linux machines, or vice-versa). But who knows!